spark/firmware
# SoftAP HTTP fixes

### Problem

See #1360

### Solution

- spark/photon-wiced#17
- Fixed chunked transfers (weren't correctly terminated with '0\r\n\r\n' empty chunk)
- Better handling of multi-packet requests
- Multiple connections support
- Fixed url routing

### Steps to Test

1. See user/tests/app/softap_http/README.md, run softap_http test
2. Check that https://docs.particle.io/reference/firmware/photon/#complete-example opens normally (it is included in the `softap_http` test application, so just navigating to http://192.168.0.1 should be enough)

### Example App

N/A

### References

Closes: #1360 
Uses: spark/photon-wiced#17

---

### Completeness


---

### Bug fix

[`[PR #1362]`](https://github.com/spark/firmware/pull/1362) [`[Fixes #1360]`](https://github.com/spark/firmware/issues/1360) Fixed SoftAP HTTP usage hard faulting in 0.7.0-rc.1 and 0.7.0-rc.2
